lib/context/form_context.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useContext, FC, ReactNode } from 'react';
+import useSWRMutation from 'swr/mutation';
+
+interface Form {
+  latLong: [number, number];
+  setFormData: (direction: number, tilt: number, capacity: number) => void;
+  setMapData: (lat: number, long: number) => void;
+}
+
+interface FormProviderProps {
+  children: ReactNode | ReactNode[];
+}
+
+type FormPostData = {
+  site_uuid: number;
+  client_name: string;
+  client_site_id: number;
+  client_site_name: string;
+  latitude: number;
+  longitude: number;
+  installed_capacity_kw: number;
+  created_utc: string;
+  updated_utc: string;
+  orientation: number;
+  tilt: number;
+};
+
+async function sendRequest(url: string, { arg }: { arg: FormPostData }) {
+  return fetch(url, {
+    method: 'POST',
+    body: JSON.stringify(arg),
+    headers: {
+      'content-type': 'application/json',
+    },
+  }).then((res) => res.json());
+}
+
+const FormContext = React.createContext<Form | null>(null);
+
+const FormProvider: FC<FormProviderProps> = ({ children }) => {
+  const { trigger } = useSWRMutation('/api/[...mockApiRoute]', sendRequest);
+
+  const [latLong, setLatLong] = useState<[number, number]>([0, 0]);
+
+  const setFormData = (direction: number, tilt: number, capacity: number) => {
+    const data: FormPostData = {
+      site_uuid: 1,
+      client_name: 'name',
+      client_site_id: 1,
+      client_site_name: 'site_name',
+      latitude: latLong[0],
+      longitude: latLong[1],
+      installed_capacity_kw: capacity,
```

Comment:
capacity rn might be NaN—which is falsy—as it is not required in our form. while this is the behavior we want, the API currently still doesn't allow for null capacity values. to get around this for now, lets put in an arbitrary constant "sentinel" value when the user doesn't have capacity specified.